awx/sso/backends.py

```diff
@@ -423,3 +463,34 @@ def on_populate_user(sender, **kwargs):
     if profile.ldap_dn != ldap_user.dn:
         profile.ldap_dn = ldap_user.dn
         profile.save()
+
+    # Get all of the orgs listed in LDAP from the database
+    roles = ['admin_role', 'auditor_role', 'member_role']
+    prefetch_roles = ["{}__members".format(role_name) for role_name in roles]
+    orgs_tied_to_ldap = Organization.objects.filter(name__in=desired_org_states.keys()).prefetch_related(*prefetch_roles)
+    # Set the state from memory into the org objects
+    for org in orgs_tied_to_ldap:
+        for role in roles:
+            if desired_org_states[org.name][role] is None:
+                # If something got a none we just need to continue on
+                pass
+            elif desired_org_states[org.name][role]:
+                if user not in getattr(org, role).members.all():
+                    logger.debug("Adding LDAP user {} to {} as {}".format(user.username, org.name, role))
+                    getattr(org, role).members.add(user)
+            elif user in getattr(org, role).members.all():
+                logger.debug("Removing LDAP user {} from {} in {}".format(user.username, role, org.name))
+                getattr(org, role).members.remove(user)
```

Comment:
What you're doing is making the code easier to follow ultimately, which the primary data structures being `desired_org_states` and `desired_team_states`, that's good :+1: 

But you should allow yourself to modify the method signatures to reflect your improved code structure. This block, implemented in a separate method, would take `desired_org_states` and `user`. That could be tested independently to verify the correctness of the database queries and updates.

Name it like `update_database_to_LDAP_orgs` or `save_LDAP_desired_org_memberships`